### PR TITLE
feat(Observable): export of, from by default

### LINKED ISSUE #894: Observable.of and Observable.from must be on Observable
Right now the Observable exported by `src/Observable` doesn't have `of` and `from` on it by default. It really should.

`from` is required for interop at the very least.

### PR DESCRIPTION
**Pull request by kwonoj**
closes #894

Now circular reference issue is resolved in TOT, this changes will allow import Observable only makes to work with `of` and `from`.

### CODE REVIEW

File: src/Rx.KitchenSink.ts [hunk 1]
```diff
@@ -30,7 +30,6 @@ import './add/observable/defer';
 import './add/observable/empty';
 import './add/observable/forkJoin';
 import './add/observable/from';
-import './add/observable/fromArray';
```
Comment (kwonoj) [inline src/Rx.KitchenSink.ts]: doesn't need to re import again since `Observable` always patches this.

File: src/Observable.ts [hunk 2]
```diff
@@ -258,3 +258,9 @@ export class Observable<T> implements CoreOperators<T>  {
   zip: <R>(...observables: Array<Observable<any> | ((...values: Array<any>) => R)>) => Observable<R>;
   zipAll: <R>(project?: (...values: Array<any>) => R) => Observable<R>;
 }
+
+//import after Observable<T> is declared and exported
+import {ArrayObservable} from './observable/fromArray';
+
+Observable.fromArray = ArrayObservable.create;
```
Comment (blesh) [inline src/Observable.ts]: We actually only need `from` and `of`. But `fromArray` is fine, since it's part of the same module anyhow.

Comment (kwonoj) [inline src/Observable.ts]: You're correct, why I did import `fromArray` only except `from`? :( Since it's same module I also think leaving `fromArray` would be fine, updating to include `from` actually..

**Comment by kwonoj** [comment]
Updated PR to patch Observable to include

`from`
`of`
`fromArray` (which based on same module with `of`, importing it at once)

**Comment by blesh** [comment]
@kwonoj unfortunately, this introduces circular dependencies to the file.

```
Observable -> observable/from
Observable -> observable/from -> observable/IteratorObservable
Observable -> observable/from -> observable/fromArray
Observable -> observable/from -> observable/fromArray -> observable/ScalarObservable
Observable -> observable/from -> observable/fromArray -> observable/ScalarObservable -> observable/empty
Observable -> observable/from -> observable/fromArray -> observable/ScalarObservable -> observable/throw
Observable -> observable/from -> observable/fromPromise
Notification -> Observable -> observable/from -> operator/observeOn-support
```

Honestly, at this point if we just rename `observable/fromArray` to `observable/of`, I think I'm okay just leaving these things out for now. We'll see how it works out.

I _am_ minorly worried about `from` though, since it's a requirement for interop.

**Comment by kwonoj** [comment]
Oops, that's bad.. 

let me rephrase to see if I understood correctly - 
```
Observable.from = ArrayObservable.create;
Observable.of = ArrayObservable.of;
```

and does not import `FromObservable` into `Observable`. is that correct?

**Comment by blesh** [comment]
@kwonoj I think this approach just won't work. 

Honestly, I'm okay with that for now. I'm cool with the modular approach of needing to import `of` and `from`.

**Comment by kwonoj** [comment]
OK, understood. I'd rather avoid of dependency issue than import too. Closing this PR.
